### PR TITLE
Experiment: dedicated menu and menu item blocks for navigation editor

### PR DESCRIPTION
**Pull request by talldan**
## Description
Something I've wanted to try for a while, is removing the dependency on the navigation block in the navigation editor.

Instead, this introduces a menu block and a menu item block, which is intended to correlate with the same functionality provided by the current WordPress menu editor.

This has been a bit of an exercise to see what gains there are from such an approach.

I've found it removes a lot of complexity.

The downside is quite a bit of code duplication, particularly for menu items / navigation links:
- Setting up variations.
- Converting pages/menu items to blocks in the placeholder.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/edit-navigation/src/constants/index.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const SIDEBAR_SCOPE = 'core/edit-navigation';
  *
  * @type {string}
  */
-export const SIDEBAR_MENU = 'edit-navigation/menu';
+export const SIDEBAR_MENU = 'core/menu';
```
Comment (talldan) [inline packages/edit-navigation/src/constants/index.js]: Oops, this shouldn't have changed.

**Comment by talldan** [comment]
Thinking about this some more, I don't think a 'Menu' block is actually needed, the editor can just use menu items directly. I might consider removing 'Menu'.

**Comment by talldan** [comment]
I've implemented the placeholder from #23953 in this PR as it was very easy to do without the technical blockers of having to extend the nav block.

**Comment by talldan** [comment]
I think this offers a lot of opportunities:
- Exploring using `useEntityProp` for editing menu item properties (similar to how a block like PostTitle works), which wouldn't be possible in the navigation link block. This architecture would make it easy to save only edited menu items.
- There has been significant feedback about the the lack of UI clarity for nested menus. This can be iterated more quickly, without having to work around how the nav block works.
- It'll be easier to show validation issues for menu items, including if there's an error from the REST API.
- The dedicated blocks have more scope for supporting backwards compatibility (especially around extensibility).
- The placeholder and block styles are already significantly improved in this PR, and it didn't take much effort.
- This PR already removes a lot of the more fragile code in the navigation editor, and there are more opportunities
- It'd be interesting to look into completely removing the 'Menu' block, thinking about it the block might be unnecessary, a user only edits menu items.
- The navigation block itself can be iterated on without concern for navigation editor support.

I think there could be some questions:
- Are these dedicated block less portable? I don't think so, as copy/paste between editors could still be handled using something like a raw transform. Or the transform system could be extended to support converting from an unsupported block to a supported one.
- How would the theme opt-in feature work? I think if the user has an existing 'Menu' block and the theme opt-in is detected, the editor could provide an option to convert/upgrade to a navigation block. The menu block also gives the user the option to stick with their classic menu if that's not what they desire.

**Comment by getdave** [comment]
Thanks for exploring this 🏅. It works well at first glance.

Here are some responses. I feel like we'll need to consider this _very_ carefully...

> * Exploring using `useEntityProp` for editing menu item properties (similar to how a block like PostTitle works), which wouldn't be possible in the navigation link block. This architecture would make it easy to save only edited menu items.

I'm not clear on how/why it's not possible? What benefit does it bring saving only edited items? Would you say this is a major win or a nice side effect?

> * There has been significant feedback about the the lack of UI clarity for nested menus. This can be iterated more quickly, without having to work around how the nav block works.

I see now that this PR drops the Nav Block _entirely_ within the Nav Editor. Is this effectively a "Classic" Navigation block?

> * It'll be easier to show validation issues for menu items, including if there's an error from the REST API.

How is it easier? I'm genuinely curious.

> * The dedicated blocks have more scope for supporting backwards compatibility (especially around extensibility).

How do they have better scope for support this? What specifically would be easier?

> * The placeholder and block styles are already significantly improved in this PR, and it didn't take much effort.

I agree the placeholder was much nicer 👍 

> * This PR already removes a lot of the more fragile code in the navigation editor, and there are more opportunities

I'm guessing this is mostly not having to disable all the "unwanted" Nav block features right?

> * It'd be interesting to look into completely removing the 'Menu' block, thinking about it the block might be unnecessary, a user only edits menu items.

I'm confused. Are you suggesting a Menu Item blocks as children of the Navigation block? 

> * The navigation block itself can be iterated on without concern for navigation editor support.

Only if we don't use the Navigation block and utilise this new Menu block instead. Otherwise we'll have to make sure Nav block changes work for the Menu block.


> * How would the theme opt-in feature work? I think if the user has an existing 'Menu' block and the theme opt-in is detected, the editor could provide an option to convert/upgrade to a navigation block. The menu block also gives the user the option to stick with their classic menu if that's not what they desire.

What happens if we decide to allow for block based themes to opt into block based rendering of menus. Wouldn't Themers now need to write styles for the Menu block as well as the Navigation block? It's another area of inconsistency and complexity to be considered.

**Comment by talldan** [comment]
> I'm not clear on how/why it's not possible? What benefit does it bring saving only edited items? Would you say this is a major win or a nice side effect?

This is something that will be necessary before shipping to core, and this way will be a much cleaner architecture, the best option we have now is to diff each menu item for changes.

> Is this effectively a "Classic" Navigation block?

Yes, but I think we're at the point where the navigation editor changes the navigation block so much that it's unrecognisable from what you see in other editors, I don't think users will perceive much of a difference in how the editor works.

> How is it easier? I'm genuinely curious.

At the moment there are two types of validation that would ideally be implemented, client side validation (catching issue before sending menu items to the REST API) and the ability to show validation errors from the REST API. The navigation block could have its own client side validation (https://github.com/WordPress/gutenberg/pull/31716), but there's no reason for this to have the same specification as what's required in the navigation editor because the block doesn't use the REST API by default. Showing errors from the server is even more difficult, the editor needs to hook into the link block which has limitations on what can be achieved. 

> How do they have better scope for support this? What specifically would be easier?

It's just a matter of having full control over the block. Whereas we want to avoid adding too many backwards compatible features to the navigation and link blocks, because its purpose is to be used in full site editing.

> I'm guessing this is mostly not having to disable all the "unwanted" Nav block features right?

Yes, there's that. The navigation block can also be cleaned up because there are a few places where the nav editor adds conditional logic to the block. The block styles are much much simpler.

> I'm confused. Are you suggesting a Menu Item blocks as children of the Navigation block?

I'm suggesting having no parent block. The navigation block in the navigation editor at the moment is entirely featureless and it adds extra UI complexity in terms of being another thing a user can select.

> Only if we don't use the Navigation block and utilise this new Menu block instead. Otherwise we'll have to make sure Nav block changes work for the Menu block.

I don't fully understand this, could you go into more detail?

> What happens if we decide to allow for block based themes to opt into block based rendering of menus. Wouldn't Themers now need to write styles for the Menu block as well as the Navigation block? It's another area of inconsistency and complexity to be considered.

The menu block would only ever render using the walker system, outputting backwards compatible HTML. Themes already support this. Additionally themes could opt in to navigation block support, which is where a themer can add additional styles. When this is detected, the editor can offer the user the option to convert from a menu block to a fully featured navigation block. But one advantage here over the current implementation is that the user can opt to keep their menu the same.

**Comment by draganescu** [comment]
I like this a lot. I think the code duplication is a minor problem in this context. As the navigation block evolves more and more towards a full fledged mega menu builder, the code duplication will be less that the code abstraction we would otherwise need on top of the navigation block to keep if from doing unsupported things or breaking.

Before we move on, what are we losing by not continuing with the navigation block as the infrastructure to edit classic menus?

**Comment by talldan** [comment]
> Before we move on, what are we losing by not continuing with the navigation block as the infrastructure to edit classic menus?

I think the main things are:
- Portability of the block needs to be addressed
- Some duplication of effort - any features implemented in the navigation block that are relevant for the navigation editor will have to be duplicated. Also the reverse of that.

**Comment by gwwar** [comment]
I really like the idea here of decoupling the blocks because there are different concerns and workflows that the navigation editor has that the other block editors do not. Trying to share code for different concerns tends to constrain how quickly we iterate (since we're adding complexity) and adds fragility as contributors are not aware of the use cases we need to handle (or when we sometimes forget to test new features in all environments).

~One really simple example is attempting to add basic transforms in https://github.com/WordPress/gutenberg/pull/34978. Because the blocks are reused by both editors, we end up needing to work around what is the block's source of truth. Navigation supports one set of transforms and the other editors need a different set. Getting this to work currently is a bit of a hack.~ Edit: it looks like this appears to work without the hook since we don't register the other block types. The transform definition not being literally what we see in the nav editor though is still confusing.

Overall:

### Navigation Editor
* Needs to handle serialization to and from menu items
* Probably will want a bigger emphasis of link variations or other UX choices to better match the older menu building UX
* Does not support newer link features (that older menu items would break on), or newer blocks.

### Post/Site/Widget Editor
* Does not need to handle serialization to and from menu items
* May need to bootstrap a navigation block from an existing menu (but this can be done as a one-way transform).
* Supports newer navigation items like search, logos, etc
* May want to diverge UX to support simpler and more modern flows. Maybe different patterns for mega menu building.

**Comment by Mamaduka** [comment]
I think @gwwar  has a good point.

In the Navigation screen, we have to have feature parity with the previous one. The Navigation block, on the other hand, doesn't need to be constrained by this.

So if the decoupling of blocks will let us iterate much faster, that's a big *pro* for me right now. Of course, we can always come back and start thinking about reusing components once both features are more stable.

**Comment by spacedmonkey** [comment]
+1 For having two blocks here. I think trying to keep the existing block generic enough to work in both places, is just going to slow both projects down ( full site editing / navigation editor ). I also agree, saddling the nav block with all the backwards compatibility constraints doesn't benefit the nav block. 

> The dedicated blocks have more scope for supporting backwards compatibility (especially around extensibility).

Consider my [comment](https://github.com/WordPress/gutenberg/issues/31551#issuecomment-918038985) here. Handling custom fields, meta boxes, filter and actions, will require a lot of complex code that is only relevant in the navigation editor. A dedicated block would help with this.

File: packages/edit-navigation/src/blocks/menu-item/index.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+/**
+ * Register the navigation link block.
+ *
+ * @uses render_block_core_navigation()
+ * @throws WP_Error An WP_Error exception parsing the block definition.
+ */
+function register_block_edit_navigation_menu_item() {
+	$post_types = get_post_types( array( 'show_in_nav_menus' => true ), 'objects' );
+	$taxonomies = get_taxonomies( array( 'show_in_nav_menus' => true ), 'objects' );
```
Comment (spacedmonkey) [inline packages/edit-navigation/src/blocks/menu-item/index.php:76]: At some point I would love to load this data via the REST API.

**Comment by draganescu** [comment]
While I was initially a big fan of having a legacy menu block, I started to wonder if this PR indicates that we've strayed too far from the original goals of the project.

Looking at our [Navigation Editor tracking issue](https://github.com/WordPress/gutenberg/issues/29102), which is currently prioritized around removing the experimental label from the project, this PR is very good at achieving that. Nevertheless, while this would help with our short term priority, it wouldn't help with our long term aims for this project.

In fact, the exploration in this PR is a prime example of why we're trying to use the navigation block in the navigation editor: to figure out, via a real world implementation, the friction points and shortcomings of: 

- the navigation block itself, 
- the block API and 
- the block editor APIs in dealing with classic WordPress data, in this case menu data.

**If making a classic menu block is the only viable solution to the above, it means we've hit quite a lot of these shortcomings, so kudos everyone, now let's improve them.**

Primarily the new block editors (widgets and navigation) started as refinement processes to power:

- increasing classic and block theme compatibility and interoperation (theme switching),
- offering a new path for block adoption in classic themes,
- improving the block API and block interface to work with reusable and structured data,
- improving the block editor APIs for working with blocks in isolation.

This approach here may unblock, and potentially speed up, the development of the navigation editor, but it has a few shortcomings:

1. It hides from us **what** the navigation block still is unable to achieve in terms of managing structured tree data editing and rendering, and also the well tested and highly dependable ways in which menus currently work in WordPress.
2. It leads to functional duplication, since in both classic and block themes menus are first and foremost the same thing: nested trees of links, so both blocks will reimplement the same kind of functionality.
3. It adds to WordPress one new special case feature, a sort of transition block, to maintain for posterity.

In conclusion, I fear if we pursue this path we’ll lose insights for improving the navigation block and the block platform as a whole, plus introduce a host of duplicate efforts. I believe we're looking to build the best block based experience possible in WordPress and this needs to push the boundaries of the platform in new and exploratory environments that only projects like the navigation editor can create. 

The development speed gains from this exploration are not worth the trade off — so I suggest let’s close this out and focus on resolving  the limitations of the navigation block and of the block platform.

**Comment by mtias** [comment]
First, thank you for doing this exploration and writing out all the perils you've identified. I wanted to chime in here because I've been against a separate block for a long time and I still don't think it's a good solution but I have a clearer perspective of where the mismatch is coming from. Or, to be more precisely, I think if we break too early we miss on a series of important architectural considerations that we need to address in the navigation block through and through.

For example:
- How does theme switching work for the block-navigation-in-a-template case?
- Can a menu be reused in different places _but customized differently_? What does that mean in block themes?
- How are the server-side APIs structured for interacting with the contents of a menu (automatically adding pages, for example)? See #30674 which is a primitive description of the same problem.
- Do block themes support automated import / display of menus and how should that behave?

In answering these I think it'd become more apparent that the characterization and the perception of a classic navigation block being intrinsically different is a bit off: 

- The navigation block got painted in the light of its most bombastic use (mega menus with mixture of presentation and IA mapping) and that was a departure that #34041 attempts to reign in.
- The different use case of this proposal largely overlaps with what the navigation block needs to accomplish in its most common incarnation if we describe it properly anyways.

@talldan's comment about considering not having a "menu" block at all I think comes from perceiving part of the problem but describing it differently. I agree we don't need it for the classic case, but I'd also argue _we don't need it for the block case_ when it comes to representing a menu structure in most situations. Let me articulate that better.

The current parent wrapper of the navigation block — `<!-- wp:navigation  -->` — is taking on the duty of handling most of the customization aspects of the block, so it represents a template concern while its contents are mostly handling the order and relationship of menu items. Conceptually, the classic navigation editor concerns itself only with the inner contents and not with the template that renders it. We should be addressing that for the navigation block case as it would also help solve some of the pending questions!

The container block should be opaque to its children when looked at from the perspective of the template that holds it. We just need to treat the inner contents as what gets saved separately and accessed independently of the navigation block container that wants to customize its look.

In other words, we need to do the conceptual shift from:

```html
<!-- wp:navigation  -->
  ...
<!-- /wp:navigation -->
```

To this for most common scenarios:

```html
<!-- wp:navigation { id: "menu" } /-->
```

The classic navigation screen concerns itself with the `...` when those are just menu links (not expanded html) in the same way the lighter navigation approach is defaulting to adding link items if there are no other blocks present. The conceptual overlap there is virtually the same and it'd be a shame to lose that.

The template that includes a menu completely eliminates the need for a "location" because that becomes an implicit aspect of use, but it does absorb configuration aspects that are decoupled from the menu structure. For example:

```html
<!-- wp:navigation {"id":"menu","isResponsive":true,"fontSize":"medium"} /-->
```

Can be used in a header template while a footer can avoid the responsive collapse and use a different font-size. The menu contents are shared.

The inner contents could be shaped in a way that is not purely relational data driven (like a mega menu) in which case reuse is mostly a matter of template part reuse when it fits.

---

I think we need to look at a next step of treating wp:navigation as an implicit template part (or different CPT, doesn't matter conceptually) first and foremost. I think some concerns will dissipate then. A dedicated screen would just be the equivalent of an isolated template part that discards the template customization to focus on the inner content arrangement. The transposition between current data and block-represented data is a much more analogous enterprise if you discard the parent for purely link-based navigations.

It's still possible that once we do these initial steps the difference that will surface is between purely relational data (link items) and menus that mix some aspects of presentation (mega menus or menus with a site logo in between items). That difference would exist in a block theme context as well, so it's not a classic vs new distinction.

**Comment by talldan** [comment]
I like the idea of a separation between the data part of a menu and the visual representation as part of making the navigation block reusable. It allows us to draw parallels between the full site editing use of the navigation block and the use in the navigation editor, which become very similar. I do think that could help improve the model that we're using.

The tough part seems to be the backwards compatibility aspect of this model. 

A question is if the navigation block should be able to read and write classic menus or whether the classic menu should be read once and then saved from that point on as a navigation block (essentially migrating data). The first way is similar to what we do now and means maintaining a lot of technical complexity, something that has been challenging. The second is cleaner, but perhaps harder to maintain back compat and also means for users switching back to the classic menu editor the story becomes interesting. The classic menu editor could probably be made to read from navigation block menus without too much effort, there's already code to do the conversion, so there are options.

**Comment by mtias** [comment]
I think two way sync compatibility would be unreasonable to maintain, as you say, and it's not a pattern we have used elsewhere before. For post content, once you go with blocks going back is not warranted to have consistent results; same has been the case for widgets. Consolidating how we store block related data is a good benefit for streamlining operations. The thing with navigation is that with this system we do have more options so we can offer more paths. For example:

- When a user "migrates" to navigation as blocks:
  - If blocks used are all link-items rendering compatibility could be 100%. Even if stored in template parts we can map the block tree to a wp_nav_menu renderer transparently without much issue.
  - If more blocks are to be used (like social icons, search, etc) a classic theme could choose to opt in by letting the block engine do the rendering (pseudo: `do_blocks( 'navigation' )`). This aligns well with the way the navigation blocks inserts links by default unless more blocks are present.
- A block editor using the "focused template part" mode for navigation would then be fine for both block and classic themes as an interface and storage.
  - On block themes the template part container would obviously handle visual display and customization while on classic or focused modes it would just represent the nav structure. Link items in this context would need to support both enabling and disabling style supports. One parallel here to consider is the "disable theme styles" we have in preferences. As block style attributes are captured in supports we can easily toggle them on or off based on what we need to display.

**Comment by getdave** [comment]
> A question is if the navigation block should be able to read and write classic menus or whether the classic menu should be read once and then saved from that point on as a navigation block (essentially migrating data)

I'd say that once the user opts into blocks for navigation _other than_ `core/navigation-link` then we make it clear (via UI prompt) that this is a lossy operation (ie: there is no two-way sync maintained). It's a one-time read and then we write to blocks from then onward.

However, if the navigation is all "simple" (just links and no other blocks) we can store as _either_ blocks or menu items. This measure of backwards compatibility will allow the navigation editor to continue to fulfil a role as a nice "onboarding ramp" for Full Site Editing whilst that project is maturing. 

In terms of data storage and how we handle transforming the different data types...I wonder if we could look to _normalize_ towards having a **single unified "schema" for navigations**.

In this world we could represent the _structure_ of a navigation in some kind of universal format (JSON or even HTML + block grammar). Then we could have different means of _persisting_ that data:

1. Persist to blocks (via Navigation block).
2. Persist to Menu CPTs (via Navigation Editor).

This would mean we'd need to create some kind of **transform layer** (`@wordpress/menus`?) that can accept _either_ blocks or classic menu data and convert it into the unified schema.

That way we'd have a Navigation block that would **only need to be aware of the universal schema** and wouldn't need to care / know about classic Menu data structures. This would also nicely reduce duplication of the existing code for transforming menu items -> blocks (and the inverse) which currently lives in _both the editor and the block._

**Comment by draganescu** [comment]
> Consolidating how we store block related data is a good benefit for streamlining operations.

Migrating to blocks could be a one way path, but IMO only when the user switches from a classic theme to a block theme.

I don't know if automatic upgrades, consolidating menu storage as block storage across theme types, are a way to go, even if the nested link blocks block fomat could be packaged to be perfectly digestible by any Walker. People may depend on the current DB menu storage, split across six tables :) I think the schema or similar approach proposed by @getdave above would be the way to find the common ground format between the JSON we get from the API and the html we get from the block storage.

**Comment by spacedmonkey** [comment]
I think that, enabling blocks in navigation, will likely end up being a opt-in feature, via a theme supports. It is unlikely that existing themes will allow blocks, like search to be put into menus, as it the css / styling would not likely "just work". So we have to consider backwards compatibility here. 

Think about these user stories. 

1. Developers of theme, does not opt in to blocks in navigation. In this case, the data will HAVE to be saved in the old format ( menu item CPT ). The user will not be able to render the blocks correctly, so let's not even try. 
2. Developers of theme, does opt in to blocks in navigation. But only has navigation links in the menu and no special block like social. For best compatibility, this should be saved as a blocks ( in menu description see https://github.com/WordPress/gutenberg/issues/34612#issuecomment-932778984 for more details ) and be saved in the old format ( menu item CPT ). This will provide the most compatibility with existing themes and will make theme switching very easy.   
3. Developers of theme, does opt in to blocks in navigation and a user saves a mixture of blocks. This is a lossy operation. This means if the user changes themes, then we can't render anything. We can add warning in wp admin, when user tries to switch theme / in menu screens. 

Again, I strongly recommend using term description to store block data, see https://github.com/WordPress/gutenberg/issues/34612#issuecomment-932778984 for more details.

**Comment by talldan** [comment]
> I think that, enabling blocks in navigation, will likely end up being a opt-in feature, via a theme supports. It is unlikely that existing themes will allow blocks, like search to be put into menus, as it the css / styling would not likely "just work". So we have to consider backwards compatibility here.

I think for non-block themes it would definitely be bad to impose block html being rendered on the frontend, I agree with that, it would very likely become a backwards compatibility mess. At the same time, the navigation editor already depends heavily on the idea of translating menu items to blocks and back again. It could be possible to make a walker to work with link blocks by converting those blocks back into the menu item format before the HTML is output, and that would have better backward compatibility results.

What would be difficult is catering for all the extensibility that can happen with menu items, things like custom fields.

> I strongly recommend using term description to store block data, see #34612 (comment) for more details.

Something that I hadn't considered at all, thanks for mentioning it. Let's continue the discussion on #34612 about that.

**Comment by draganescu** [comment]
> the navigation editor already depends heavily on the idea of translating menu items to blocks

Just to crosslink my comment https://github.com/WordPress/gutenberg/issues/34612#issuecomment-933003760 where I suggest we move this transforming concern out of the block.

**Comment by tellthemachines** [comment]
>  On block themes the template part container would obviously handle visual display and customization

To make sure we're all talking about the same thing here 😅 , would the "template part container" in this case be the Navigation block? Asking because we've been discussing these ideas [here](https://github.com/WordPress/gutenberg/issues/30007#issuecomment-933129709) and are unclear on that point.

> Link items in this context would need to support both enabling and disabling style supports

Wouldn't the current setup where the Navigation contents inherit context from their parent work? The link items themselves shouldn't have any style controls, but if styling is present in their container they use it.

**Comment by talldan** [comment]
> To make sure we're all talking about the same thing here 😅 , would the "template part container" in this case be the Navigation block

@tellthemachines My understanding is that would be the case, but I don't really see the connection to that comment of mine.

**Comment by tellthemachines** [comment]
> @tellthemachines My understanding is that would be the case, but I don't really see the connection to that comment of mine.

This discussion is becoming super confusing 😅 

You were replying to my question on whether the nav editor would be using the Navigation block or just its contents. Because the Nav block has all the styling options etc, there really isn't anything in the block itself that we need to use in the editor, at least data-wise.

 I may not have been fully clear with my question above. I should have asked instead:

> On block themes the template part container would obviously handle visual display and customization while on classic or focused modes it would just represent the nav structure

Is the template part container used in block themes **the same** as the one used in "classic or focused modes"? 

Because I took `<!-- wp:navigation { id: "menu" } /-->` to mean the menu, as contents of the nav block, is coming in from somewhere else (a custom post type), and we can grab that menu and use it in the nav editor, independently of the nav block.

**Comment by getdave** [comment]
> Because I took `<!-- wp:navigation { id: "menu" } /-->` to mean the menu, as contents of the nav block, is coming in from somewhere else (a custom post type), and we can grab that menu and use it in the nav editor, independently of the nav block.

Yes this is what I currently believe.

* `core/navigation` - wrapper concerned with presentational purposes. Akin to Template Part.
* `core/navigation-link` (and friends!) - inner blocks representing the structure/data of a "navigation".

The implication for the Nav Editor is that it no longer utilises the `core/navigation` block at all. Rather it acts like an Isolated Template Part editor and contains only the child blocks of the Navigation.

This is very similar to the existing classic Menus screen which is only concerned with the structure/data of the navigation and not the _presentation_ of the Navigation.
